### PR TITLE
feat: add contract verifier

### PR DESCRIPTION
**Pull request by mrice32**
The idea behind this contract is primarily to stop users from depositing ETH into a SpokePool address that does not exist on a particular chain -- this can happen if the wallet and dapp have different ideas about which chain the transaction is destined for.

### CODE REVIEW

File: contracts/SpokePoolVerifier.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./interfaces/SpokePoolInterface.sol";
+
+/**
+ * @notice SpokePoolVerifier is a contract that verifies that the SpokePool exists on this chain before sending ETH to it.
+ */
+contract SpokePoolVerifier {
+    using Address for address;
+
+    function deposit(
```
Comment (james-a-morris) [inline contracts/SpokePoolVerifier.sol:13]: Should we add natspec docs to this?

**Review by james-a-morris — APPROVED** [review]
Only a small nit.

File: deploy/023_deploy_spoke_pool_verifier.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  await deploy("SpokePoolVerifier", {
+    from: deployer,
+    log: true,
+    skipIfAlreadyDeployed: true,
+    deterministicDeployment: "0x1234", // Salt for the create2 call.
+  });
+};
+
+module.exports = func;
+func.tags = ["SpokePoolVerifier", "polygon", "mainnet", "optimism", "arbitrum", "zksync"];
```
[the remark this answers is not in the document]
Comment (mrice32) [reply deploy/023_deploy_spoke_pool_verifier.ts:19]: I think there is a native token!

File: contracts/SpokePoolVerifier.sol (hunk 1 above)
Comment (mrice32) [reply contracts/SpokePoolVerifier.sol:13]: yes!

**Comment by mrice32** [comment]
@james-a-morris @pxrl I will fix your comments in a follow-up!

**Pull request merged**